modules/margin/src/main/java/com/opengamma/sdk/margin/MarginWhatIfCalcResult.java

```diff
@@ -0,0 +1,506 @@
+/*
+ * Copyright (C) 2017 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.sdk.margin;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+import org.joda.beans.impl.direct.DirectPrivateBeanBuilder;
+
+/**
+ * Result from the service.
+ */
+@BeanDefinition(builderScope = "private", metaScope = "private", factoryName = "of")
+public final class MarginWhatIfCalcResult implements ImmutableBean {
+
+  /**
+   * The status of the calculation - PENDING or COMPLETED.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final MarginCalcResultStatus status;
+  /**
+   * The type of calculation.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final MarginCalcRequestType type;
+  /**
+   * The valuation date for which the portfolio will be processed.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LocalDate valuationDate;
+  /**
+   * The reporting currency, as an ISO 4217 three letter currency code.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final String reportingCurrency;
+  /**
+   * The summary of the portfolio items, may be empty.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final List<PortfolioItemSummary> portfolioItems;
+  /**
+   * The result of the margin calculation
+   */
+  @PropertyDefinition(get = "optional")
+  private final MarginWhatIfSummary margin;
```

Comment:
Would it be better to return the `MarginSummary` from each call, and a smaller combined result?